File: src/main/api/payment.ts
/**
 * Payment API
 * Handles payment and credit operations
 */

import {
  AvailableCurrency,
  CreatePaymentRequest,
  CreatePaymentResponse,
  CreditPlan,
  PaymentHistory,
  PaymentStatusResponse,
} from '../types/payment.js';
import { ApiClient, ApiResponse } from './client.js';

export class PaymentApi extends ApiClient {
  /**
   * Get available credit plans
   */
  async getPlans(): Promise<
    ApiResponse<{ plan: CreditPlan; credits: number; price_usd: number; popular: boolean }[]>
  > {
    return this.get('/api/payment/plans');
  }

  /**
   * Get available currencies for payment
   */
  async getAvailableCurrencies(): Promise<ApiResponse<AvailableCurrency[]>> {
    return this.get('/api/payment/currencies');
  }

  /**
   * Create a new payment
   */
  async createPayment(data: CreatePaymentRequest): Promise<ApiResponse<CreatePaymentResponse>> {
    return this.post('/api/payment/create', data);
  }

  /**
   * Get payment status
   */
  async getPaymentStatus(paymentId: string): Promise<ApiResponse<PaymentStatusResponse>> {
    return this.get(`/api/payment/status/${paymentId}`);
  }

  /**
   * Get payment history
   */
  async getPaymentHistory(): Promise<ApiResponse<PaymentHistory[]>> {
    return this.get('/api/payment/history');
  }

  /**
   * Get current user credits
   */
  async getCredits(): Promise<ApiResponse<{ credits: number }>> {
    return this.get('/api/payment/credits');
  }
}

File: src-tauri/src/services/payment.rs
use serde_json::Value;

use crate::services::api_client::ApiClient;
use crate::store::ConfigStore;

pub struct PaymentService;

impl PaymentService {
    fn client(config_store: &ConfigStore) -> ApiClient {
        let token = config_store.get_config().session_token;
        if token.is_empty() { ApiClient::new() } else { ApiClient::new().with_token(token) }
    }

    pub async fn get_plans(config_store: &ConfigStore) -> Result<Value, String> {
        Self::client(config_store).get("/api/payment/plans").await
    }

    pub async fn get_currencies(config_store: &ConfigStore) -> Result<Value, String> {
        Self::client(config_store).get("/api/payment/currencies").await
    }

    pub async fn create_payment(config_store: &ConfigStore, data: Value) -> Result<Value, String> {
        Self::client(config_store).post("/api/payment/create", &data).await
    }

    pub async fn get_payment_status(config_store: &ConfigStore, payment_id: &str) -> Result<Value, String> {
        Self::client(config_store)
            .get(&format!("/api/payment/status/{}", payment_id)).await
    }

    pub async fn get_payment_history(config_store: &ConfigStore) -> Result<Value, String> {
        Self::client(config_store).get("/api/payment/history").await
    }

    pub async fn get_credits(config_store: &ConfigStore) -> Result<Value, String> {
        Self::client(config_store).get("/api/payment/credits").await
    }
}
